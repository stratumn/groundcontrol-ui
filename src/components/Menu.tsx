// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import graphql from "babel-plugin-relay/macro";
import React, { Fragment } from "react";
import { createFragmentContainer } from "react-relay";
import {
  Container,
  Menu as SemanticMenu,
  Responsive,
  Sidebar,
} from "semantic-ui-react";

import { Menu_system } from "./__generated__/Menu_system.graphql";

import MenuPrimaryItems from "./MenuPrimaryItems";
import MenuSecondaryItems from "./MenuSecondaryItems";

export interface IProps {
  system: Menu_system;
  showSidebar: boolean;
  onShowSidebar: (values: IProps) => any;
  onHideSidebar: (values: IProps) => any;
}

export function Menu(props: IProps) {
  const {
    system: {
      serviceMetrics,
      jobMetrics,
      processMetrics,
      logMetrics,
    },
    showSidebar,
    onShowSidebar,
    onHideSidebar,
  } = props;

  const handleShowSidebar = () => onShowSidebar({...props});
  const handleHideSidebar = () => onHideSidebar({...props});

  return (
    <Fragment>
      <SemanticMenu
        fixed="top"
        size="large"
        color="teal"
        inverted={true}
      >
        <Container fluid={true}>
          <Responsive
            as={SemanticMenu.Item}
            icon="bars"
            maxWidth={767}
            onClick={handleShowSidebar}
          />
          <Responsive
            as={MenuPrimaryItems}
            minWidth={768}
          />
          <Responsive
            as={SemanticMenu.Menu}
            position="right"
            minWidth={768}
          >
            <MenuSecondaryItems
              serviceMetrics={serviceMetrics}
              jobMetrics={jobMetrics}
              processMetrics={processMetrics}
              logMetrics={logMetrics}
            />
          </Responsive>
        </Container>
      </SemanticMenu>
      <Responsive
        as={Fragment}
        maxWidth={767}
      >
        <Sidebar
          as={SemanticMenu}
          vertical={true}
          size="large"
          animation="push"
          visible={showSidebar}
          onHide={handleHideSidebar}
        >
          <MenuPrimaryItems />
          <MenuSecondaryItems
            serviceMetrics={serviceMetrics}
            jobMetrics={jobMetrics}
            processMetrics={processMetrics}
            logMetrics={logMetrics}
          />
        </Sidebar>
      </Responsive>
    </Fragment>
  );
}

export default createFragmentContainer(Menu, graphql`
  fragment Menu_system on System {
    jobMetrics {
      ...MenuSecondaryItems_jobMetrics
    }
    serviceMetrics {
      ...MenuSecondaryItems_serviceMetrics
    }
    processMetrics {
      ...MenuSecondaryItems_processMetrics
    }
    logMetrics {
      ...MenuSecondaryItems_logMetrics
    }
  }`,
);
