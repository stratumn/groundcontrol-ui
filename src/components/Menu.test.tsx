// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from "enzyme";
import React from "react";

import { mockQueryPropAttrs } from "../testing/relay";

import { Menu } from "./Menu";

const props = {
  onHideSidebar: jest.fn(),
  onShowSidebar: jest.fn(),
  showSidebar: true,
  system: {
    ...mockQueryPropAttrs(),
    jobMetrics: {
      ...mockQueryPropAttrs(),
    },
    logMetrics: {
      ...mockQueryPropAttrs(),
    },
    processMetrics: {
      ...mockQueryPropAttrs(),
    },
    serviceMetrics: {
      ...mockQueryPropAttrs(),
    },
  },
};

describe("<Menu />", () => {

  it("renders correctly", () => {
    const wrapper = shallow(<Menu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

});
