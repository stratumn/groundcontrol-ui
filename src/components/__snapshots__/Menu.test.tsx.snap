// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders correctly 1`] = `
<Fragment>
  <Menu
    color="teal"
    fixed="top"
    inverted={true}
    size="large"
  >
    <Container
      fluid={true}
    >
      <Responsive
        as={[Function]}
        getWidth={[Function]}
        icon="bars"
        maxWidth={767}
        onClick={[Function]}
      />
      <Responsive
        as={[Function]}
        getWidth={[Function]}
        minWidth={768}
      />
      <Responsive
        as={[Function]}
        getWidth={[Function]}
        minWidth={768}
        position="right"
      >
        <ForwardRef(Relay(MenuSecondaryItems))
          jobMetrics={
            Object {
              " $fragmentRefs": null,
              " $refType": null,
            }
          }
          logMetrics={
            Object {
              " $fragmentRefs": null,
              " $refType": null,
            }
          }
          processMetrics={
            Object {
              " $fragmentRefs": null,
              " $refType": null,
            }
          }
          serviceMetrics={
            Object {
              " $fragmentRefs": null,
              " $refType": null,
            }
          }
        />
      </Responsive>
    </Container>
  </Menu>
  <Responsive
    as={Symbol(react.fragment)}
    getWidth={[Function]}
    maxWidth={767}
  >
    <Sidebar
      animation="push"
      as={[Function]}
      direction="left"
      duration={500}
      onHide={[Function]}
      size="large"
      vertical={true}
      visible={true}
    >
      <MenuPrimaryItems />
      <ForwardRef(Relay(MenuSecondaryItems))
        jobMetrics={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
          }
        }
        logMetrics={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
          }
        }
        processMetrics={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
          }
        }
        serviceMetrics={
          Object {
            " $fragmentRefs": null,
            " $refType": null,
          }
        }
      />
    </Sidebar>
  </Responsive>
</Fragment>
`;
